website/config/routes.js

```diff
@@ -280,6 +280,7 @@ module.exports.routes = {
   'GET /get-started':            '/fleetctl-preview',
   'GET /g':                       (req,res)=> { let originalQueryStringWithAmp = req.url.match(/\?(.+)$/) ? '&'+req.url.match(/\?(.+)$/)[1] : ''; return res.redirect(301, sails.config.custom.baseUrl+'/?meet-fleet'+originalQueryStringWithAmp); },
   'GET /test-fleet-sandbox':     '/try-fleet/register',
+  'GET /unsubscribe':            { action: 'unsubscribe-from-all-newsletters' },
```

Comment:
Since this isn't a redirect, it's an API endpoint that gets triggered through the URL bar (like /logout), we should move it there and apply the same approach as we did w/ logout; i.e.:
- make it have an api-looking route for clarity
- but also add a redirect for `/unsubscribe`, like we do for `/logout` (id put it in the same place and handle it the same way)

Reply:
<img width="613" alt="image" src="https://user-images.githubusercontent.com/618009/186856758-eef44742-c2d8-4176-8c13-866f4fc05b6b.png">

Reply:
> but also add a redirect for /unsubscribe, like we do for /logout (id put it in the same place and handle it the same way)

@mikermcneil Would this type of redirect (`'GET /unsubscribe': '/api/v1/unsubscribe-from-all-newsletters'`) work with an input passed in from query string?
It looks like we would have to do something similar to the redirect function for `/g` to pass the query string into the action https://github.com/fleetdm/fleet/blob/f5e4dc47dfe822d5d9ff3e52ab2f174af16af442/website/config/routes.js#L281

Although I'm wondering if it wouldn't be better to handle this in a different way. With this current behavior, users unsubscribing would be taken to an API endpoint that would just display "OK" when they unsubscribe. 

How would you feel about adding the code from `unsubscribe-from-all-newsletters.js` inline to `view-homepage-or-redirect` and directing users who want to unsubscribe (through the link at the bottom of the newsletter email layout) to `https://fleetdm.com/?unsubscribe=[emailAddress]` instead?

Reply:
Good point.  

How about instead of concerning ourselves with the parameters, we update the defaults for the action so that "unsubscribe from all" is the default.

You hit /unsubscribe, it redirects you, triggering the action which responds such that you see "OK", and you're fully unsubscribed.

I think users who visit the unsubscribe URL should see a page letting them know that it has worked, but that can be a quick follow-up PR / issue and needn't block merging this, since there won't actually be release notes going out yet via email.  cc @mike-j-thomas

Reply:
>How about instead of concerning ourselves with the parameters, we update the defaults for the action so that "unsubscribe from all" is the default.

The only input of that action is the emailAddress of the NewsletterSubscription record that is being updated. 

>You hit /unsubscribe, it redirects you, triggering the action which responds such that you see "OK", and you're fully unsubscribed.

I've updated the redirect to pass the query string to the API endpoint.

Reply:
> emailAddress

ah, yes, this makes sense, thanks.